### ISSUE TITLE
Nhipster blob entity generator issue front angular

### ISSUE THREAD

**Issue by Toysurusse** [jhipster/generator-jhipster-nodejs#201]
##### **Overview of the issue**
How to fix Blob front Angular view issue ?

##### **Motivation for or Use Case**
When you use nhipster entity generator with an angular front. Angular can't render the image of the blob after beeing storage in database.

##### **Reproduce the error**
use nhipster entity generator and you will have issue to view blob from backend.

##### **Suggest a Fix**
when the service nest get the blob image from the server, the image is convert in an buffer of hexadecimal. So I had to fix the service to add a converter hexaToString like the following function :
    const blob = roadMap.image.toString('hex');
    var result = '';
    for (var i = 0; i < blob.length; i = i + 2) {
      var decval = parseInt(blob.substr(i, 2), 16);
      result = result + String.fromCharCode(decval);
    }

##### **JHipster Version(s)**
 last version

##### **JHipster configuration**
nhipster

**Comment by avdev4j** [comment]
@amanganiello90 I have moved this ticket to the proper repository ;)

**Comment by amanganiello90** [comment]
Thanks.

**Comment by jlopezjuy** [comment]
> ##### **Overview of the issue**
> How to fix Blob front Angular view issue ?
> 
> ##### **Motivation for or Use Case**
> When you use nhipster entity generator with an angular front. Angular can't render the image of the blob after beeing storage in database.
> 
> ##### **Reproduce the error**
> use nhipster entity generator and you will have issue to view blob from backend.
> 
> ##### **Suggest a Fix**
> when the service nest get the blob image from the server, the image is convert in an buffer of hexadecimal. So I had to fix the service to add a converter hexaToString like the following function :
> const blob = roadMap.image.toString('hex');
> var result = '';
> for (var i = 0; i < blob.length; i = i + 2) {
> var decval = parseInt(blob.substr(i, 2), 16);
> result = result + String.fromCharCode(decval);
> }
> 
> ##### **JHipster Version(s)**
> last version
> 
> ##### **JHipster configuration**
> nhipster

I have same error, because nest with typeor get the blob file from database on BUFFER type. and this sugested fix only work on determinates files. im using files.

@Column({ type: 'blob', name: 'documento', nullable: true })
  documento: any;

  @Column({ name: 'documento_content_type', nullable: true })
  documentoContentType: string;
  
  on mi service:
  
const blob = gestion.documento.toString('hex');
        var result = '';
        for (var i = 0; i < blob.length; i = i + 2) {
          var decval = parseInt(blob.substr(i, 2), 16);
          result = result + String.fromCharCode(decval);
        }
        console.log(result);
        gestion.documento = result;
        docuFiltered.push(gestion);

**Comment by amanganiello90** [comment]
@jlopezjuy thanks for the response, but I don't understand very well your message. You are saying, that the fix suggested do not resolve the angular blob issue?

**Comment by jlopezjuy** [comment]
> @jlopezjuy thanks for the response, but I don't understand very well your message. You are saying, that the fix suggested do not resolve the angular blob issue?

yes, if i use blop to sabe pdf or other file type, the solution not work.

**Comment by amanganiello90** [comment]
> 
> 
> > @jlopezjuy thanks for the response, but I don't understand very well your message. You are saying, that the fix suggested do not resolve the angular blob issue?
> 
> yes, if i use blop to sabe pdf or other file type, the solution not work.

But @Toysurusse  suggests that fix resolved the blob issue. So I don't understand :(

**Comment by jlopezjuy** [comment]
maybe the solution work only with images... not with files...

**Comment by amanganiello90** [comment]
Ahh ok, you say that if the blob is an image the fix works, instead if the blob is a generic file (for example a .txt) it does not work. Right?

**Comment by jlopezjuy** [comment]
yes

**Comment by jlopezjuy** [comment]
i was tested with diferent pdf and other files.

**Comment by amanganiello90** [comment]
@jlopezjuy  thanks. Have you some suggestions to fix in order to work with every blob type?

**Comment by jlopezjuy** [comment]
On java jhipster app this work ok because spring return blob into base64, typorm return Buffer... maybe thats is the problem...



> @jlopezjuy thanks. Have you some suggestions to fix in order to work with every blob type?

mmm not yet, I keep trying to solve the bug, because I need a project that works with pdf
